plugins/inputs/prometheus/prometheus.go

```diff
@@ -360,9 +367,71 @@ func (p *Prometheus) Gather(acc telegraf.Accumulator) error {
 	return nil
 }
 
-func (p *Prometheus) gatherURL(u URLAndAddress, acc telegraf.Accumulator) error {
+func setResult(resultString string, requestFields map[string]interface{}, tags map[string]string) {
+	resultCodes := map[string]int{
+		"success":                 0,
+		"unable_to_decode":        1,
+		"body_read_error":         2,
+		"connection_failed":       3,
+		"timeout":                 4,
+		"dns_error":               5,
+		"http_code_not_ok":        6,
+		"error_reading_body":      7,
+		"content_length_exceeded": 8,
+	}
+
+	tags["result"] = resultString
+	requestFields["result_code"] = resultCodes[resultString]
```

Comment:
Translating the error string to a numeric value is up to the user and this should not be hard-coded.

The result string can continue to exist as a tag, but the result code field should be dropped.

Reply:
I understand but I use the same method as here:
https://github.com/influxdata/telegraf/blob/master/plugins/inputs/http_response/http_response.go#L141